### PR TITLE
 History.txt -> History.md, add links & reformat

### PR DESCRIPTION
**Pull request by MSP-Greg**
Change History.txt to History.md.  All listed pulls/issues are now links, and the code snippets are highlighted and indented correctly.  Left original text file.

I was in another repo with a similar issue, so I wrote some code to generate the links.  I'll post it somewhere.

Not sure if it will render as markdown here, see:
https://github.com/MSP-Greg/rake-compiler/blob/history-txt-2-md/History.md

### CODE REVIEW

**Comment by kou** [comment]
Sorry for my late response.
Could you remove `History.txt`? We can't maintain both `History.txt` and `History.md`.

We also need to update `rake-compiler.gemspec`. (We may have more files to be updated.)
So please don't add `[ci skip]`.

**Comment by DapperBlondie** [comment]
زبون بسته چي مينالي تو😂😂😂😂😂😂😂😂

Alireza Gharib
Electrical Engineering Student
Iran's University of Science & Technology
________________________________

**Comment by DapperBlondie** [comment]
ببند زبون بسته نفله نکبت😂😂😂😂😂

Alireza Gharib
Electrical Engineering Student
Iran's University of Science & Technology

________________________________

**Comment by MSP-Greg** [comment]
@kou

I'll remove the [ci skip] and delete History.txt.

Looking at the file, the links clutter it, I can place them at the bottom of the doc?  Might be a day or two.

.gemspec has `s.files += Dir.glob("*.{md,rdoc,txt,yml}")` , so no issue with the History files?

**Comment by kou** [comment]
We can keep the inline links.
We'll not be able to maintain separated links when we add a new release entry.

https://github.com/rake-compiler/rake-compiler/blob/master/rake-compiler.gemspec#L43 may have a problem.

**Comment by olleolleolle** [comment]
Hey, this is a ✨ sparkling improvement ✨ and I love it. I think that even if the newer releases are not "fully compliant" with links and all, that it would be easy enough to add that after-the-fact for each release. The release cadence is not _that_ high, is it?

File: History.md [hunk 1]
```diff
@@ -155,134 +155,140 @@
...
   * Add support for cross-builds and multiple platforms (x86/x64).
-    Pull #74 [larskanis]
+    Pull [#74](https://github.com/rake-compiler/rake-compiler/pull/74) [larskanis]
 
-      $ rake-compiler cross-ruby VERSION=1.8.7-p371
-      $ rake-compiler cross-ruby VERSION=1.9.3-p392
-      $ rake-compiler cross-ruby VERSION=2.0.0-p0
-      $ rake-compiler cross-ruby VERSION=2.0.0-p0 HOST=x86_64-w64-mingw32
-      $ rake cross compile RUBY_CC_VERSION=1.8.7:1.9.3:2.0.0
+    ```ruby
```
Comment (kou) [inline History.md:184]:
```suggestion
    ```text
```

**Pull request merged**